### ISSUE TITLE
not able log to wandb 

### ISSUE THREAD

**Issue by Lalit-001** [ThilinaRajapakse/simpletransformers#528]
hey, I am **trying to log my model training to wandb**. So i followed the given tutorial in the documentation. but it **not able to set link b/w wandb config parameter and model hyperparameters**. 
Because 

1. I **initialize num_train_epochs = 2 in sweep_config**  but **my model training staring with num_train_epochs =1**
2. [wandb chart](https://app.wandb.ai/00archer0/multilabel-tuning/sweeps/b10hr6mz?workspace=user-00archer0) eval_loss metric is always null.

could you help me with code? if I am anything wrong.

```
sweep_config = {
    
    "method": "bayes",  # grid, random
    "metric": {"name": "eval_loss", "goal": "minimize"},
    "parameters": {
        "num_train_epochs": {"values": [2, 3, 5]},
        "learning_rate": {"min": 5e-5, "max": 4e-4},
    },
}

sweep_id = wandb.sweep(sweep_config,project="multilabel-tuning")
```

```
model_args = {
    'train_batch_size':64,
    'eval_batch_size':8,
    'reprocess_input_data': True,
    'overwrite_output_dir': True,
    'no_save': True,
    "use_multiprocessing": True,
    'fp16': False,
    'do_lower_case': True,
  #  'num_train_epochs': 3,
    'max_seq_length': 64,
    'regression': False,
    'manual_seed': 3,
  #  "learning_rate":4e-5,
    'weight_decay':0,
    "save_eval_checkpoints": False,
    "save_model_every_epoch": False,
    "silent": False,
    "evaluate_during_training": True,
    "evaluate_during_training_verbose": True,
    
}
```

```
def train_mod(train1_trn,train1_val):
    # Initialize a new wandb run
    wandb.init(config=sweep_config, id=sweep_id)
    

    # Create a TransformerModel
    model = MultiLabelClassificationModel('bert', 'bert-large-uncased-whole-word-masking', 
                                          use_cuda=True,
                                          num_labels=4,
                                          args=model_args,
                                          sweep_config=wandb.config)

    # Train the model
    model.train_model(train1_trn,
                      eval_df=train1_val,

                      )
    
    
    # Evaluate the model
    model.eval_model(train1_val)

    # Sync wandb
    wandb.join()
```

```
fold=StratifiedKFold(n_splits=8, shuffle=True, random_state=3)

for train_index, test_index in fold.split(train1,train1['target']):
    train1_trn, train1_val = train1.iloc[train_index], train1.iloc[test_index]

    train1_trn = label_encode(train1_trn)
    train1_val = label_encode(train1_val)
    print("Train: ",train1_trn.shape,"Test: ",train1_val.shape)

    

    wandb.agent(sweep_id, train_mod(train1_trn,train1_val))
    break
```
```


Train:  (539, 2) Test:  (77, 2)

Logging results to Weights & Biases (Documentation).
Project page: https://app.wandb.ai/00archer0/multilabel-tuning
Run page: https://app.wandb.ai/00archer0/multilabel-tuning/runs/b10hr6mz

INFO:wandb.run_manager:system metrics and metadata threads started
INFO:wandb.run_manager:checking resume status, waiting at most 10 seconds
INFO:wandb.run_manager:resuming run from id: UnVuOnYxOmIxMGhyNm16Om11bHRpbGFiZWwtdHVuaW5nOjAwYXJjaGVyMA==
INFO:wandb.run_manager:upserting run before process can begin, waiting at most 10 seconds
INFO:wandb.run_manager:saving pip packages
INFO:wandb.run_manager:initializing streaming files api
INFO:wandb.run_manager:unblocking file change observer, beginning sync with W&B servers
INFO:wandb.run_manager:file/dir modified: /content/wandb/run-20200701_142033-b10hr6mz/config.yaml
INFO:wandb.run_manager:file/dir created: /content/wandb/run-20200701_142033-b10hr6mz/wandb-events.jsonl
ERROR:wandb.run_manager:Streaming file created twice in same run: /content/wandb/run-20200701_142033-b10hr6mz/wandb-events.jsonl
INFO:wandb.run_manager:file/dir created: /content/wandb/run-20200701_142033-b10hr6mz/wandb-summary.json
INFO:wandb.run_manager:file/dir created: /content/wandb/run-20200701_142033-b10hr6mz/wandb-metadata.json
INFO:wandb.run_manager:file/dir created: /content/wandb/run-20200701_142033-b10hr6mz/wandb-history.jsonl
INFO:wandb.run_manager:file/dir created: /content/wandb/run-20200701_142033-b10hr6mz/requirements.txt
WARNING:transformers.modeling_utils:Some weights of the model checkpoint at bert-large-uncased-whole-word-masking were not used when initializing BertForMultiLabelSequenceClassification: ['cls.predictions.bias', 'cls.predictions.transform.dense.weight', 'cls.predictions.transform.dense.bias', 'cls.predictions.transform.LayerNorm.weight', 'cls.predictions.transform.LayerNorm.bias', 'cls.predictions.decoder.weight', 'cls.seq_relationship.weight', 'cls.seq_relationship.bias']
- This IS expected if you are initializing BertForMultiLabelSequenceClassification from the checkpoint of a model trained on another task or with another architecture (e.g. initializing a BertForSequenceClassification model from a BertForPretraining model).
- This IS NOT expected if you are initializing BertForMultiLabelSequenceClassification from the checkpoint of a model that you expect to be exactly identical (initializing a BertForSequenceClassification model from a BertForSequenceClassification model).
WARNING:transformers.modeling_utils:Some weights of BertForMultiLabelSequenceClassification were not initialized from the model checkpoint at bert-large-uncased-whole-word-masking and are newly initialized: ['classifier.weight', 'classifier.bias']
You should probably TRAIN this model on a down-stream task to be able to use it for predictions and inference.
INFO:simpletransformers.classification.classification_model: Converting to features started. Cache is not used.

100%
539/539 [00:00<00:00, 1303.36it/s]


Epoch 1 of 1: 100%
1/1 [00:12<00:00, 12.87s/it]
Running Epoch 0: 100%
9/9 [00:11<00:00, 1.33s/it]

Running loss: 0.740408

INFO:wandb.run_manager:file/dir modified: /content/wandb/run-20200701_142033-b10hr6mz/wandb-metadata.json

Running loss: 0.390839

INFO:simpletransformers.classification.classification_model: Converting to features started. Cache is not used.


INFO:simpletransformers.classification.classification_model:{'LRAP': 0.7445887445887446, 'eval_loss': 0.4505506843328476}
INFO:simpletransformers.classification.classification_model: Training of bert model complete. Saved to outputs/.


100%
77/77 [00:00<00:00, 1041.92it/s]


Running Evaluation: 100%
10/10 [00:18<00:00, 1.80s/it]


INFO:wandb.run_manager:file/dir modified: /content/wandb/run-20200701_142033-b10hr6mz/wandb-events.jsonl
INFO:wandb.wandb_agent:Running runs: []
INFO:wandb.wandb_agent:Agent received command: run
INFO:wandb.wandb_agent:Agent starting run with config:
	learning_rate: 5.138402535892588e-05
	num_train_epochs: 2
INFO:wandb.wandb_agent:About to run command: /usr/bin/env python  --learning_rate=5.138402535892588e-05 --num_train_epochs=2

wandb: Agent Starting Run: ktv4fd59 with config:
	learning_rate: 5.138402535892588e-05
	num_train_epochs: 2

INFO:wandb.run_manager:file/dir modified: /content/wandb/run-20200701_142033-b10hr6mz/wandb-metadata.json
INFO:wandb.wandb_agent:Running runs: ['ktv4fd59']
INFO:wandb.wandb_agent:Cleaning up finished run: ktv4fd59
INFO:wandb.wandb_agent:Agent received command: run
INFO:wandb.wandb_agent:Agent starting run with config:
	learning_rate: 0.00024326253544768448
	num_train_epochs: 5
INFO:wandb.wandb_agent:About to run command: /usr/bin/env python  --learning_rate=0.00024326253544768448 --num_train_epochs=5

wandb: Agent Starting Run: 0jnlbuc0 with config:
	learning_rate: 0.00024326253544768448
	num_train_epochs: 5

INFO:wandb.wandb_agent:Running runs: ['0jnlbuc0']
INFO:wandb.wandb_agent:Cleaning up finished run: 0jnlbuc0
INFO:wandb.wandb_agent:Agent received command: run
INFO:wandb.wandb_agent:Agent starting run with config:
	learning_rate: 0.0002534776366127239
	num_train_epochs: 3
INFO:wandb.wandb_agent:About to run command: /usr/bin/env python  --learning_rate=0.0002534776366127239 --num_train_epochs=3

wandb: Agent Starting Run: i6u9w7jh with config:
	learning_rate: 0.0002534776366127239
	num_train_epochs: 3

INFO:wandb.run_manager:file/dir modified: /content/wandb/run-20200701_142033-b10hr6mz/wandb-metadata.json
INFO:wandb.wandb_agent:Running runs: ['i6u9w7jh']
2020-07-01 14:21:23,558 - wandb.wandb_agent - ERROR - Detected 3 failed runs in the first 60 seconds, shutting down.
ERROR:wandb.wandb_agent:Detected 3 failed runs in the first 60 seconds, shutting down.
INFO:wandb.wandb_agent:To disable this check set WANDB_AGENT_DISABLE_FLAPPING=true
INFO:wandb.run_manager:shutting down system stats and metadata service

CPU times: user 18.1 s, sys: 7.07 s, total: 25.1 s
Wall time: 49.6 s

INFO:wandb.run_manager:file/dir modified: /content/wandb/run-20200701_142033-b10hr6mz/wandb-events.jsonl
INFO:wandb.run_manager:stopping streaming files and file change observer
INFO:wandb.run_manager:file/dir modified: /content/wandb/run-20200701_142033-b10hr6mz/wandb-metadata.json

```

**Comment by ThilinaRajapakse** [comment]
Does the example in the [docs](https://simpletransformers.ai/docs/usage/#hyperparameter-optimization) work as expected for you?

**Comment by Lalit-001** [comment]
neither that example is working.

**Comment by Lalit-001** [comment]
I just followed that because it was in documentation, so I thought after some hit trial it will work. but.......

**Comment by ThilinaRajapakse** [comment]
There was a mistake in the example code (missing line) which has now been fixed. The problem is that the `wandb_project` must be specified for simple transformers to automatically log metrics and this line was missing in the example.

**Comment by Lalit-001** [comment]
Thank you, for the update, it's a great fix, will save some time to setup wandb. like I did manage to do it in the traditional way.

```
import wandb

from simpletransformers.classification import (
    MultiLabelClassificationModel
)

import logging
logger = logging.getLogger("wandb")
logger.setLevel(logging.ERROR)

logging.basicConfig(level=logging.INFO)
transformers_logger = logging.getLogger("transformers")
transformers_logger.setLevel(logging.WARNING)

import pandas as pd

train_args = {
    'train_batch_size':32,
    'eval_batch_size':8,
    'reprocess_input_data': True,
    'overwrite_output_dir': True,
    'no_save': True,
    "use_multiprocessing": True,
    'fp16': False,
    'do_lower_case': True,
    'num_train_epochs': 1,
    'max_seq_length': 64,
    'regression': False,
    'manual_seed': 3,
   "learning_rate": 0.0004,
    'weight_decay':0,
    "save_eval_checkpoints": False,
    "save_model_every_epoch": False,
    "silent": True,
    "evaluate_during_training": True,
    "evaluate_during_training_verbose": True,

}

def train_mod(train1_trn,train1_val):
  
    train_args["learning_rate"] = wandb.config.learning_rate
    train_args["num_train_epochs"] = wandb.config.num_train_epochs
    train_args["eval_batch_size"] = wandb.config.eval_batch_size
    train_args["train_batch_size"] = wandb.config.train_batch_size

    # Create a TransformerModel
    model = MultiLabelClassificationModel('bert', 'bert-large-uncased-whole-word-masking', 
                                          use_cuda = True,
                                          num_labels=4,
                                          )

    # Train the model
    model.train_model(train1_trn,
                      eval_df=train1_val,
                      args=train_args,
                      )

    # Evaluate the model
    result, model_outputs, wrong_predictions = model.eval_model(train1_val)
    
    return result
    

train1_trn = pd.read_json("/content/train1_trn.json",)
train1_val = pd.read_json("/content/train1_val.json")


# Pass your defaults to wandb.init
wandb.init(config=train_args)
config = wandb.config

result = train_mod(train1_trn,train1_val)
wandb.log(result)
```

but by this fix it will be more genalize and fast.
